### ISSUE TITLE
Install failure "You cannot add the same module/version combo more than once" on fresh install of Pi OS Bookworm on Pi Model 1 B?

### ISSUE THREAD

**Issue by Jibun-no-Kage** [morrownr/rtl8852bu#29]
Attempted installation of driver, as noted in the title, then tried remove script, and tried install script again.  Same result as below.

`: ---------------------------
: remove-driver.sh v20231118
: armv6l (kernel architecture)
: 6.6.28+rpt-rpi-v6 (kernel version)
: ---------------------------

--> rtl8852bu 1.19.3: added
Error! The module/version combo: rtl8852bu-1.19.3: is not located in the DKMS tree.
rtl8852bu/1.19.3 has been removed
Removing 8852bu.conf from /etc/modprobe.d
Removing source files from /usr/src/rtl8852bu-1.19.3
The driver was removed successfully.
You may now delete the driver directory if desired.
: ---------------------------

Do you want to reboot now? (recommended) [Y/n] y
`
`
# ./install-driver.sh
: ---------------------------
: install-driver.sh v20231115
: armv6l (kernel architecture)
: arm (architecture to send to gcc)
: 1/1 (in-use/total processing units)
: 438192 (total system memory)
: 6.6.28+rpt-rpi-v6 (kernel version)
: gcc-12 (version of gcc used to compile the kernel)
: gcc (Raspbian 12.2.0-14+rpi1) 12.2.0
: dkms-3.0.10
: mokutil not installed
: ---------------------------

Checking for previously installed drivers...

--> rtl8852bu 1.19.3: added
Error! The module/version combo: rtl8852bu-1.19.3: is not located in the DKMS tree.
Finished checking for and removing previously installed drivers.
: ---------------------------

Starting installation.
Installing 8852bu.conf to /etc/modprobe.d
The dkms installation routines are in use.
Copying source files to /usr/src/rtl8852bu-1.19.3
Error! DKMS tree already contains: rtl8852bu-1.19.3
You cannot add the same module/version combo more than once.
This driver may already be installed.
Run the following and then reattempt installation.
$ sudo ./remove-driver.sh
`

Pi OS information...

'
  `.::///+:/-.        --///+//-:``    root@garage
 `+oooooooooooo:   `+oooooooooooo:    ---------------------------------
  /oooo++//ooooo:  ooooo+//+ooooo.    OS: Raspbian GNU/Linux 12 (bookworm) armv6l
  `+ooooooo:-:oo-  +o+::/ooooooo:     Host: Raspberry Pi Model B Rev 2
   `:oooooooo+``    `.oooooooo+-      Kernel: 6.6.28+rpt-rpi-v6
     `:++ooo/.        :+ooo+/.`       Uptime: 7 mins
        ...`  `.----.` ``..           Packages: 775 (dpkg)
     .::::-``:::::::::.`-:::-`        Shell: bash 5.2.15
    -:::-`   .:::::::-`  `-:::-       Resolution: 1024x768
   `::.  `.--.`  `` `.---.``.::`      Terminal: /dev/pts/0
       .::::::::`  -::::::::` `       CPU: BCM2835 (1) @ 700MHz
 .::` .:::::::::- `::::::::::``::.    Memory: 50MiB / 427MiB
-:::` ::::::::::.  ::::::::::.`:::-
::::  -::::::::.   `-::::::::  ::::
-::-   .-:::-.``....``.-::-.   -::-
 .. ``       .::::::::.     `..`..
   -:::-`   -::::::::::`  .:::::`
   :::::::` -::::::::::` :::::::.
   .:::::::  -::::::::. ::::::::
    `-:::::`   ..--.`   ::::::.
      `...`  `...--..`  `...`
            .::::::::::
             `.-::::-`
'
`
# cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 12 (bookworm)"
NAME="Raspbian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
`
`
# uname -a
Linux garage 6.6.28+rpt-rpi-v6 #1 Raspbian 1:6.6.28-1+rpt1 (2024-04-22) armv6l GNU/Linux
`

**Comment by Jibun-no-Kage** [comment]
More details...

```
# cat /proc/version
Linux version 6.6.28+rpt-rpi-v6 (debian-kernel@lists.debian.org) (gcc-12 (Raspbian 12.2.0-14+rpi1) 12.2.0, GNU ld (GNU Binutils for Raspbian) 2.40) #1 Raspbian 1:6.6.28-1+rpt1 (2024-04-22)

# gcc --version
gcc (Raspbian 12.2.0-14+rpi1) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by morrownr** [comment]
Hi @Jibun-no-Kage 

There you go breaking my driver. Actually I have a patch to the scripts that should correct the problem of the "added" driver but now I am curious what took place to cause the problem. I'll try to add the patch today.

In the title, you say "You cannot add the same module/version combo more than once". Are you trying to run two adaptors with the rtl8852bu chipset on one computer?

**Comment by morrownr** [comment]
I've applied some patches. Try:

$ git pull
$ sudo sh install-driver.sh

**Comment by morrownr** [comment]
@Jibun-no-Kage 

It seems like I remember that you have adapters based on the rtl8821/11cu and/or the rtl8821/11au chips. If so, there is a new driver in testing to go into the kernel for the rtl8821/11au and we have made big improvements to the rtl8821/11cu. Let me know if you are interested in testing.

**Comment by Jibun-no-Kage** [comment]
Actually I was just loading the specific driver to prepare a new Pi OS base image, using ansible and it tripped over the error.  I then was able to recreate the issue manually, so thought I would document it.  And I now have a few newer wifi adapters.  As for doing additional testing, sure, glad to help.

**Comment by morrownr** [comment]
I had found a bug in the install script when a driver is only added to dkms so I had to do a lot of work to fix that situation. The new code was only in the rtl8812au driver and has been testing but I think it is good. The problem should be fixed.

For adapters with the rtl8821/11au chip:

https://github.com/morrownr/8821au-20210708/issues/133

Do you also have a rtl8821/11cu based adapter?

**Comment by Jibun-no-Kage** [comment]
Yes, I believe I have 8821/11 based device(s), I will confirm.  The 8852bu I believe is the most recent device I have.  I have not updated my routers to WIFI 6 as yet, so that is also future goal.  But given what I have been reading about the ups and downs with WIFI 6 in come cases... maybe I can live with out WIFI 6 for a while? :)

**Comment by morrownr** [comment]
> I believe I have 8821/11 based device(s), 

Let me know what you have and I can point you in the right direction for testing.

> The 8852bu I believe is the most recent device I have.

I'm not a fan of the Realtek WiFi 6 generation of usb adapters and drivers. Bad support from Realtek. For WiFi 6 and 7, usb users are better off going with Mediatek based adapters. I have many listed in the Plug and Play List.

> maybe I can live with out WIFI 6 for a while? 

Recommend you take a look at the below link. It is about OpenWRT One. It is a new wifi router designed by OpenWRT and produced by Banana Pi. It is in testing but should be available soon. My WiFi 6 router uses the same SoC as OpenWRT One so I know this can be a very good reasonable price product.

https://forum.banana-pi.org/t/openwrt-one-ap-24-xy-is-an-upcoming-router-board-developed-by-openwrt-and-banana-pi/17047
